src/EditorFeatures/Core/Implementation/DocumentationComments/AbstractDocumentationCommentCommandHandler.cs

```diff
@@ -89,11 +89,10 @@ private static void ApplySnippet(DocumentationCommentSnippet snippet, ITextBuffe
             }
 
             var service = document.GetRequiredLanguageService<IDocumentationCommentSnippetService>();
-            var syntaxTree = document.GetRequiredSyntaxTreeSynchronously(cancellationToken);
-            var text = syntaxTree.GetText(cancellationToken);
             var documentOptions = document.GetOptionsAsync(cancellationToken).WaitAndGetResult(cancellationToken);
-
-            var snippet = getSnippetAction(service, syntaxTree, text, caretPosition, documentOptions, cancellationToken);
+            var semanticModel = document.GetPartialSemanticModelAsync(cancellationToken).WaitAndGetResult(cancellationToken);
```

Comment:
tbh, this is just incredibly concerning.  semantic models (and the work therin) is not cheap, and this is a sync codepath.  do we need semantics?  can't we look for throws purely syntactically?